erpnext/hooks.py

```diff
@@ -249,6 +249,9 @@
 		"on_update": [
 			"erpnext.support.doctype.service_level_agreement.service_level_agreement.update_hold_time",
 			"erpnext.support.doctype.issue.issue.set_first_response_time"
+		],
+		"after_insert": [
+			"erpnext.crm.doctype.lead.lead.add_prospect_link_in_communication"
```

Comment:
This should not happen on every communication insert.

We need to do the following:

For each lead in the prospect child table, add the prospect’s link in the "**Links**" table of all the address and contacts linked to that Lead. This will allow rendering all address and contact details in the prospect.

**When should we trigger adding prospect's link to leads' contacts and addresses?** 

- While grouping leads into prospects (prospect server-side update)
- Whenever we add a new contact or address in the lead that is linked to a prospect, add the prospect link to those address and contacts.

![image](https://user-images.githubusercontent.com/24353136/130663639-b5198677-381c-4fe9-a5be-29820e842e07.png)

Linking prospect with contact will also show all the linked emails in the Prospect’s timeline

You can read more about email aggregation here:

https://erpnext-documentation.frappe.cloud/docs/v13/user/manual/en/setting-up/email/linking-emails-to-document#1-email-aggregation-for-customer-and-supplier